### PR TITLE
Add coppi and merckx autoyast profile (kernel QA)

### PR DESCRIPTION
**Pull request by pevik**
Profile is based on autoyast_mlx_con5.xml (sonic & tails profile usedfor InfiniBand testing) with these changes:

* removed sections: `<bootloader>`, `<firewall>`, `<networking>`, `<software><image>`
* hostname (sonic-1) and MAC address for eth0 (ac:1f:6b:44:01:10), both part of <networking>): define hostname and MAC in profile shared on more machines is simply wrong (although eth0 might not be used (at least not for IB tests on sonic & tails).
* trailing whitespace

Problems fixed:

* `Extra element firewall in interleave<br>42:0: ERROR: Element profile failed to validate contentNode.`
* `Extra element networking in interleave<br>356:0: ERROR: Element profile failed to validate contentNode.`
* `Extra element bootloader in interleave<br>24:0: ERROR: Element profile failed to validate contentNode.`
* According to bsc#1176595#c7 also <software><image> section was wrong.

NOTE: this profile works also for sonic and tails (i.e. SLES installation works), but these are used for InfiniBand testing, which requires more setup (autoyast_mlx_con5.xml needs to be fixed).

Related ticket: https://progress.opensuse.org/issues/59024#note-5

Verification run: https://openqa.suse.de/tests/4760354

### CODE REVIEW

**Comment by pevik** [comment]
WIP: I removed other sections, because it keeps failing. TODO: verify it.

**Comment by schlad** [comment]
Thanks for working on this. Changes in such xml file are not going to be easy to understand in the future, I mean when looking at the git log, so more meaningful git commit message would be great.

**Comment by pevik** [comment]
Sure. Now it's not valid because of prototyping. It started as removal of single section. But it looks like half of the profile isn't valid, thus I'll try to create minimal config, inspired by `data/autoyast_sle15/mini.xml`.

**Comment by pevik** [comment]
It looks like the problem isn't in the profile https://bugzilla.suse.com/show_bug.cgi?id=1176595 (**this PR should not be merged**).

**Comment by pevik** [comment]
It looks this version is working, at least for coppi.
https://openqa.suse.de/tests/4759806#live
I'll check it also for sonic and tails.

**Comment by pevik** [comment]
Works also for sonic and tails: https://openqa.suse.de/tests/4759980 https://openqa.suse.de/tests/4759981

**Comment by schlad** [comment]
Quick question: I did not follow much that bug, but isn't that the elements you remove are in fact a valid product bugs? https://bugzilla.suse.com/show_bug.cgi?id=1176595#c5

I'm not sure if simple removal of all those elements won't break IB tests?

Question: wouldn't it be better to have more ay profiles one could use depending on what is needed for specific tests?
I'm simply worry that we merge this now and on Monday @frankenmichl will simply need to restore quite some of those elements.

Sorry if above questions are annoying.

**Comment by pevik** [comment]
Sure, we can wait for Michie.

My few cents: In LTP we don't rely on autoyast setup, because we need to solve these things also for qemu. For me it's safer to have minimal autoyast profile to get machine installed and then do the "setup" manually than solving all over again broken installation (half of the time installation on IPMI was broken; valid bugs: I'm not a part of functional team, I need to have installed system to test other parts than yast).
Back to possible failures on sonic and tails: let's wait for the result: https://openqa.suse.de/tests/4759980 https://openqa.suse.de/tests/4759981. If it breaks, I'd like to reuse this config at least for LTP (thus rename it to use it for coppi and merckx only). 
BTW looking at coppi (installed with https://openqa.suse.de/tests/4759806): hostname is set correctly to coppi-1, thus it looks like machine specific setup is not needed to get correct hostname (I'd try to avoid machine specific setup unless needed, we don't want to maintain several files).

**Comment by schlad** [comment]
> Sure, we can wait for Michie.
> 
> My few cents: In LTP we don't rely on autoyast setup, because we need to solve these things also for qemu. For me it's safer to have minimal autoyast profile to get machine installed and then do the "setup" manually than solving all over again broken installation 

That is exactly what I am proposing. Sorry if I wasn't clear.
I meant - instead of modifying the big ay we have now (which has mlx in the name), provide a separate, simplistic one.
Hope I am more clear now. Sorry again.

**Comment by pevik** [comment]
TL;DR
I'll add this profile as `data/autoyast_sle15/autoyast_64bit-epyc-eth25G.xml`, thus it's used only for coppi and merckx.
And leave sonic & tails for you and Michie to fix, ok?

More thoughts: one thing I don't understand: should we have:
1) machine specific autoyast profile, or
2) testsuite specific autoyast profile, or
3) something in between, i.e. autoyast profile testsuite specific for some machine groups
IMHO 3. would be needed, because 
* I see some failures in https://openqa.suse.de/tests/4759980#live => something removed is needed for IB tests, but worked for LTP => specific testsuite changes needed.
* But also specific machine setup (frodo and tyrion needs different partitioning than these 4 machines).

**Comment by schlad** [comment]
> TL;DR
> I'll add this profile as `data/autoyast_sle15/autoyast_64bit-epyc-eth25G.xml`, thus it's used only for coppi and merckx.
> And leave sonic & tails for you and Michie to fix, ok?

This sounds indeed very good!

> 
> More thoughts: one thing I don't understand: should we have:
> 
>     1. machine specific autoyast profile, or
> 
>     2. testsuite specific autoyast profile, or
> 
>     3. something in between, i.e. autoyast profile testsuite specific for some machine groups
>        IMHO 3. would be needed, because

Hmmm I do not work so much with those yet, but from my point of view, we should have machine specific ay profiles. After all underlying hardware plays a role here.
At the same time for some tests you might not care for setting up NICs, so I could imagine that for machine X we might have ay profile X.simplistic X.with_NIC X_with_HBA etc.
After all your point of simplifying ay profiles is super valid and if some tests you run do not need to have for instance NICs configured, you might be better off by having super simple ay for machine X but still for testsuite Y.

Does it make sense from point of view?

> 
> 
>     * I see some failures in https://openqa.suse.de/tests/4759980#live => something removed is needed for IB tests, but worked for LTP => specific testsuite changes needed.

yeah I was expecting something like this :/ 

> 
>     * But also specific machine setup (frodo and tyrion needs different partitioning than these 4 machines).

**Comment by pevik** [comment]
Michie also proposes having machine specific configs: https://progress.opensuse.org/issues/59024#note-5
Thus I'll update title and comment of this PR and solve this for coppi (code already changed). Unfortunately you need to fix sonic & tails, but at least you have a hint what is working in `data/autoyast_sle15/autoyast_64bit-epyc-eth25G.xml`.

**Comment by schlad** [comment]
> Unfortunately you need to fix sonic & tails, but at least you have a hint what is working in `data/autoyast_sle15/autoyast_64bit-epyc-eth25G.xml`.

yeah thanks for your work here. It will be now so much easier to fix ay for sonic and tails!

**Comment by schlad** [comment]
LGTM!

**Review by schlad — APPROVED** [review]
